src/main.cpp

```diff
@@ -5183,7 +5187,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             LOCK(cs_main);
             State(pfrom->GetId())->fProvidesHeaderAndIDs = !(nLocalServices & NODE_WITNESS) || (nCMPCTBLOCKVersion == 2);
             State(pfrom->GetId())->fPreferHeaderAndIDs = fAnnounceUsingCMPCTBLOCK;
-            State(pfrom->GetId())->fWantCmpctWitness = (nCMPCTBLOCKVersion == 2);
+            State(pfrom->GetId())->fSupportsCmpctWitness = (nCMPCTBLOCKVersion == 2);
```

Comment:
@sipa  I guess that's fine.  I noted the language I thought could be clarified to make this more clear in the BIP.

I agree that ignoring v2 messages if we're not NODE_WITNESS should result in the identical behavior we're looking for.